Agents/EmailAgent/email-agent-code/src/main/java/uk/ac/cam/cares/jps/agent/email/EmailAgentConfiguration.java

```diff
@@ -0,0 +1,159 @@
+     */
+    public static final String KEY_SMTP_PASS = "smtp.pass";
+
+    /**
+     * Key for SMTP port setting.
+     */
+    public static final String KEY_SMTP_PORT = "smtp.port";
+
+    /**
+     * Key for SSL setting.
+     */
+    public static final String KEY_SSL_ENABLE = "smtp.ssl.enable";
+
+    /**
+     * Key for SMTP auth setting.
+     */
+    public static final String KEY_SMTP_AUTH = "smtp.auth";
+
+    /**
+     * Key for Subject prefix setting.
+     */
+    public static final String KEY_SUBJECT_PREFIX = "subject.prefix";
+
+    /**
+     * Key for destination address.
+     */
+    public static final String KEY_TO_ADDRESS = "to.address";
+
+    /**
+     * Key for source address.
+     */
+    public static final String KEY_FROM_ADDRESS = "from.address";
+
+    /**
+     * Key for white list status.
+     */
+    public static final String KEY_WHITE_ONLY = "whitelist.enabled";
+
+    /**
+     * Key for white list IPs
+     */
+    public static final String KEY_WHITE_IPS = "whitelist.ips";
+
+    /**
+     * Live properties object.
+     */
+    private static Properties properties;
+
+    /**
+     * Returns the property represented by the input key as a string (or null if not found).
+     *
+     * @param key property key
+     *
+     * @return property value (or null)
+     *
+     * @throws IllegalStateException if properties has not been read
+     */
+    static String getProperty(String key) {
+        if (properties == null) {
+            throw new IllegalStateException("Properties file has not been loaded!");
+        }
+
+        return properties.getProperty(key, null);
+    }
+
+    /**
+     * Returns the property represented by the input key as token separated array (or null if not
+     * found).
+     *
+     * @param key property key
+     * @param delimiter token to split on
+     *
+     * @return property value array (or null)
+     *
+     * @throws IllegalStateException if properties has not been read
+     */
+    static String[] getPropertyAsArray(String key, String delimiter) {
+        if (properties == null) {
+            throw new IllegalStateException("Properties file has not been loaded!");
+        }
+
+        if (properties.containsKey(key)) {
+            return properties.getProperty(key).split(delimiter);
+        }
+        return null;
+    }
+
+    /**
+     * Reads the default properties file.
+     *
+     * @throws IOException if properties file cannot be read.
+     */
+    static void readProperties() throws IOException {
+        properties = new Properties();
+
+        String propertyFileLocation = System.getenv(PROPERTIES_ENV);
+        System.out.println("Attempting to read properties file at: " + propertyFileLocation);
+        
+        try ( FileInputStream file = new FileInputStream(propertyFileLocation)) {
+            properties.load(file);
+            
+            boolean whitelistOn = Boolean.parseBoolean(getProperty(KEY_WHITE_ONLY));
```

Comment:
Default, if `KEY_WHITE_ONLY` not set, is `false` is that intended?

Reply:
You're correct, if not set the "whitelistOn" variable should be false, will resolve.

Reply:
Default if not set _should_ be false.